packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -85,123 +109,506 @@ const propTypes = {
+        setInputFocus(event, dayInputRef.current, date.day.length, date.day.length);
+      } else if (event.keyCode === KEY_RIGHT && monthInputRef.current.selectionEnd === date.month.length) {
+        setInputFocus(event, yearInputRef.current, 0, 0);
+      }
+    } else if (dateFormatOrder === DateUtil.dateOrder.YMD) {
+      if ((event.keyCode === KEY_LEFT || event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE) && monthInputRef.current.selectionEnd === 0) {
+        setInputFocus(event, yearInputRef.current, date.year.length, date.year.length);
+      } else if (event.keyCode === KEY_RIGHT && monthInputRef.current.selectionEnd === date.month.length) {
+        setInputFocus(event, dayInputRef.current, 0, 0);
+      }
+    }
+  };
+
+  const handleYearInputKeydown = (event) => {
+    if (dateFormatOrder === DateUtil.dateOrder.MDY) {
+      if ((event.keyCode === KEY_LEFT || event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE) && yearInputRef.current.selectionEnd === 0) {
+        setInputFocus(event, dayInputRef.current, date.day.length, date.day.length);
+      }
+    } else if (dateFormatOrder === DateUtil.dateOrder.DMY) {
+      if ((event.keyCode === KEY_LEFT || event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE) && yearInputRef.current.selectionEnd === 0) {
+        setInputFocus(event, monthInputRef.current, date.month.length, date.month.length);
+      }
+    } else if (dateFormatOrder === DateUtil.dateOrder.YMD) {
+      if (event.keyCode === KEY_RIGHT && yearInputRef.current.selectionEnd === date.year.length) {
+        setInputFocus(event, monthInputRef.current, 0, 0);
+      }
+    }
+  };
+
+  const handleOnInputFocus = (event, type) => {
+    if (onFocus) {
+      onFocus(event);
+    }
+
+    setFocused(true);
+
+    if (type === DateUtil.inputType.DAY) {
+      setDayInitialFocused(true);
+      dayInputRef.current.setSelectionRange(0, dayInputRef.current.value.length);
+    } else if (type === DateUtil.inputType.MONTH) {
+      setMonthInitialFocused(true);
+      monthInputRef.current.setSelectionRange(0, monthInputRef.current.value.length);
+    } else {
+      setYearInitialFocused(true);
+      yearInputRef.current.setSelectionRange(0, yearInputRef.current.value.length);
+    }
+  };
+
+  const handleOnInputBlur = (event, type) => {
+    if (onBlur) {
+      onBlur(event);
+    }
+
+    setFocused(false);
+
+    if (type === DateUtil.inputType.DAY) {
+      setDayInitialFocused(false);
+    } else if (type === DateUtil.inputType.MONTH) {
+      setMonthInitialFocused(false);
+    } else {
+      setYearInitialFocused(false);
+    }
+
+    if (type === DateUtil.inputType.DAY || type === DateUtil.inputType.MONTH) {
+      let inputValue = event.target.value;
+
+      // Prepend a 0 to the value when losing focus and the value is single digit.
+      if (inputValue.length === 1) {
+        inputValue = inputValue === '0' ? '' : '0'.concat(inputValue);
```

Comment:
Why do we want to delete `0` if that is the entry. It seems a bit annoying that it gets deleted when the user may come back to finish the entry.

Reply:
Makes sense, updated here : https://github.com/cerner/terra-framework/pull/1208/commits/623265a07fb2323e62a269e3e72b183ee8c8b2da